pystiche/papers/ulyanov_et_al_2016/loss.py

```diff
@@ -0,0 +1,167 @@
+
+def ulyanov_et_al_2016_content_loss(
+    impl_params: bool = True,
+    instance_norm: bool = True,
+    multi_layer_encoder: Optional[MultiLayerEncoder] = None,
+    layer: str = "relu_4_2",
+    score_weight=None,
+):
+    if score_weight is None:
+        if impl_params:
+            score_weight = 1e0 if instance_norm else 6e-1
+        else:
+            score_weight = 1e0
+
+    if multi_layer_encoder is None:
+        multi_layer_encoder = ulyanov_et_al_2016_multi_layer_encoder()
+    encoder = multi_layer_encoder[layer]
+
+    return MSEEncodingOperator(encoder, score_weight=score_weight)
+
+
+class UlyanovEtAl2016GramOperator(GramOperator):
+    def __init__(
+        self, encoder: Encoder, impl_params: bool = True, **gram_op_kwargs: Any,
+    ):
+        super().__init__(encoder, **gram_op_kwargs)
+        self.normalize_by_num_channels = impl_params
+
+    def enc_to_repr(self, enc: torch.Tensor) -> torch.Tensor:
+        gram_matrix = super().enc_to_repr(enc)
+        if not self.normalize_by_num_channels:
+            return gram_matrix
+
+        num_channels = gram_matrix.size()[-1]
+        return gram_matrix / num_channels
+
+
+def ulyanov_et_al_2016_style_loss(
+    impl_params: bool = True,
+    instance_norm: bool = True,
+    stylization: bool = True,
+    multi_layer_encoder: Optional[MultiLayerEncoder] = None,
+    layers: Optional[Sequence[str]] = None,
+    layer_weights: Union[str, Sequence[float]] = "sum",
+    score_weight: float = None,
+    **gram_op_kwargs,
+):
+    if score_weight is None:
+        if impl_params:
+            if instance_norm:
+                score_weight = 1e0
+            else:
+                score_weight = 1e3 if stylization else 1e0
+        else:
+            score_weight = 1e0
+
+    if multi_layer_encoder is None:
+        multi_layer_encoder = ulyanov_et_al_2016_multi_layer_encoder()
+
+    if layers is None:
+        if impl_params:
+            layers = (
+                ("relu_1_1", "relu_2_1", "relu_3_1", "relu_4_1")
+                if instance_norm
+                else ("relu_1_1", "relu_2_1", "relu_3_1", "relu_4_1", "relu_5_1")
+            )
+        else:
+            layers = ("relu_1_1", "relu_2_1", "relu_3_1", "relu_4_1", "relu_5_1")
+
+    def get_encoding_op(encoder, layer_weight):
+        return UlyanovEtAl2016GramOperator(
+            encoder, score_weight=layer_weight, **gram_op_kwargs
+        )
+
+    return MultiLayerEncodingOperator(
+        multi_layer_encoder,
+        layers,
+        get_encoding_op,
+        layer_weights=layer_weights,
+        score_weight=score_weight,
+    )
+
+
+class UlyanovEtAl2016PerceptualLoss(MultiOperatorLoss):
+    def __init__(
+        self,
+        style_loss: MultiLayerEncodingOperator,
+        content_loss: MSEEncodingOperator = None,
+        stylization: bool = True,
+    ) -> None:
+
+        dict = OrderedDict([("style_loss", style_loss)])
+        if stylization:
+            dict["content_loss"] = content_loss
```

Comment:
There is no need to pass `stylization` here. You could simply check `if content_loss is not None`.